sources/packages/backend/apps/api/src/route-controllers/application-offering-change-request/application-offering-change-request.students.controller.ts

```diff
@@ -0,0 +1,136 @@
+   * @param studentUserToken student user token to authorize the user.
+   * @returns Application Offering Change Request details.
+   */
+  @Get(":applicationOfferingChangeRequestId")
+  @ApiNotFoundResponse({
+    description: "Not able to find an Application Offering Change Request.",
+  })
+  async getById(
+    @Param("applicationOfferingChangeRequestId", ParseIntPipe)
+    applicationOfferingChangeRequestId: number,
+    @UserToken()
+    studentUserToken: StudentUserToken,
+  ): Promise<ApplicationOfferingDetailsAPIOutDTO> {
+    return this.applicationOfferingChangeRequestControllerService.getById(
+      applicationOfferingChangeRequestId,
+      {
+        studentId: studentUserToken.studentId,
+        applicationOfferingDetails: true,
+      },
+    );
+  }
+
+  /**
+   * Gets the Application Offering Change Request status.
+   * @param applicationOfferingChangeRequestId the Application Offering Change Request id.
+   * @param studentUserToken student user token to authorize the user.
+   * @returns Application Offering Change Request status.
+   */
+  @Get(
+    ":applicationOfferingChangeRequestId/application-offering-change-request-status",
+  )
+  @ApiNotFoundResponse({
+    description:
+      "Not able to get the Application Offering Change Request Status.",
+  })
+  async getApplicationOfferingChangeRequestStatusById(
+    @Param("applicationOfferingChangeRequestId", ParseIntPipe)
+    applicationOfferingChangeRequestId: number,
+    @UserToken()
+    studentUserToken: StudentUserToken,
+  ): Promise<ApplicationOfferingChangeRequestStatusAPIOutDTO> {
+    const applicationOfferingChangeRequestStatus =
+      await this.applicationOfferingChangeRequestService.getApplicationOfferingChangeRequestStatusById(
+        applicationOfferingChangeRequestId,
+        {
+          studentId: studentUserToken.studentId,
+        },
+      );
+    if (!applicationOfferingChangeRequestStatus) {
+      throw new NotFoundException(
+        "Not able to get the Application Offering Change Request Status.",
+      );
+    }
+    return { status: applicationOfferingChangeRequestStatus };
+  }
+
+  /**
+   * Updates an application offering change request status.
+   * @param applicationOfferingChangeRequestId application offering change request id.
+   * @param userToken user token to authorize the user.
+   * @param payload information to update the application offering change request status.
+   */
+  @Patch(":applicationOfferingChangeRequestId")
+  @ApiUnauthorizedResponse({
+    description:
+      "The student does not have access to the application offering change request.",
+  })
+  async updateApplicationOfferingChangeRequestStatus(
+    @Param("applicationOfferingChangeRequestId", ParseIntPipe)
+    applicationOfferingChangeRequestId: number,
+    @UserToken()
+    userToken: StudentUserToken,
+    @Body()
+    payload: StudentApplicationOfferingChangeRequestAPIInDTO,
+  ): Promise<void> {
+    const studentAuthorized =
```

Comment:
The variable name `studentAuthorized` doesn't seem be right. It could be just student.